app/lib/features/space/dialogs/edit_space_sheet.dart

```diff
@@ -241,11 +241,33 @@ class _EditSpacePageConsumerState extends ConsumerState<EditSpacePage> {
               );
               return;
             }
+            // check permissions before updating space
+            bool havePermission = await permissionCheck();
+            if (!havePermission) {
+              popUpDialog(
+                context: context,
+                title: Padding(
+                  padding: const EdgeInsets.all(8.0),
+                  child: Icon(
+                    Atlas.block_prohibited,
+                    size: 28,
+                    color: Theme.of(context).colorScheme.onError,
+                  ),
+                ),
+                subtitle: const Text('Cannot edit space with no permissions'),
+                btn2Text: 'Okay',
+                onPressedBtn2: () => context.pop(),
+                btn2Color: Theme.of(context).colorScheme.success,
+              );
+              return;
+            }
             final roomId = await _handleUpdateSpace(context);
             debugPrint('Space Updated: $roomId');
             // refresh spaces and side bar
             ref.invalidate(spacesProvider);
             ref.invalidate(nav.spaceItemsProvider);
+            ref.invalidate(spaceProvider);
```

Comment:
This will forcefully refresh the providers until the next frame loads .i.e. after navigation